scripts/activity_detection_exploration/support_functions.py

```diff
@@ -0,0 +1,249 @@
+    slice_index_ranges: List[Tuple[int, int]] = []
+
+    # The [start, end) frame time pairs
+    slice_time_ranges: List[Tuple[float, float]] = []
+
+    # Prediction results per slice
+    slice_preds: List[Dict[str, float]] = []
+
+    @classmethod
+    def clear_slice_values(cls):
+        """ Clear variable states with new list instances. """
+        cls.slice_index_ranges = []
+        cls.slice_time_ranges = []
+        cls.slice_preds = []
+
+
+@dataclasses.dataclass
+class SelectedSlice:
+    index: int
+    animation: animation.FuncAnimation
+
+    @property
+    def frame_sequence(self) -> np.ndarray:
+        """ matrix of image frames, shape [nFrames x H x W x C] """
+        slice_idx_range = GlobalValues.slice_index_ranges[self.index]
+        slice_frames = GlobalValues.all_image_mats[slice_idx_range[0]:slice_idx_range[1]]
+        return slice_frames
+
+    @property
+    def activity_predictions(self):
+        return GlobalValues.slice_preds[self.index]
+
+
+def plot_activity_confidence(label, gt_ranges, custom_range=None, custom_range_color=None):
+    """
+    Plot activity confidences, with hover-over showing the sequence-middle frame.
+    Clicks modify the object output to be used for
+
+    :param label: String label of the activity class predictions to render.
+    :param gt_ranges: A sequence of tuples indicating the starting and ending time of ground-truth
+                      time ranges the label activity occurred in the image sequence.
+    :param custom_range: Optional tuple indicating the starting and ending times of an additional
+                         range to highlight in addition to the `gt_ranges`.
+    :param custom_range_color: The color of the additional range to be drawn. If not set, we will
+                               use "red".
+
+    Learned from: http://www.andrewjanowczyk.com/image-popups-on-mouse-over-in-jupyter-notebooks/
+    """
+    fig = plt.figure(figsize=(14, 6))
+    ax = fig.add_subplot(111)
+    ax.set_title(f"Window Confidence over time for \"{label}\"")
+    ax.set_xlabel("Time")
+    ax.set_ylabel("Confidence")
+    ax.set_ylim(0, 1.05)
+    ax.set_xlim(GlobalValues.slice_time_ranges[0][0]-1,
+                GlobalValues.slice_time_ranges[-1][1]+1)
+    # plt.yscale("log")
+
+    # Bar plt to show bars where the "true" time ranges are for the activity.
+    xs_bars = [p[0] for p in gt_ranges]
+    ys_gt_regions = [1 for _ in gt_ranges]
+    bar_widths = [(p[1]-p[0]) for p in gt_ranges]
+    ax.bar(xs_bars, ys_gt_regions, width=bar_widths, align="edge", color="lightgreen")
+
+    if custom_range:
+        assert len(custom_range) == 2, "Assuming only two float values for custom range"
+        xs_bars2 = [custom_range[0]]
+        ys_height = [1.025] #[0.1]
+        bar_widths2 = [custom_range[1]-custom_range[0]]
+        ys_bottom = [0] #[1.01]
+        color = custom_range_color if custom_range_color is not None else "red"
```

Comment:
Can we just set the custom_range_color="red" in the function definition?

Reply:
Valid. I suspect this was an artifact of my hacking things quickly before this was added as part of this function.